lib/src/facts/solaris/virtualization_resolver.cc

```diff
@@ -26,65 +36,37 @@ namespace facter { namespace facts { namespace solaris {
             return vm::zone;
         }
 
-        auto isa = facts.get<string_value>(fact::hardware_isa);
-        if (!isa) {
-            return {};
-        }
-
         string guest_of;
 
-        if (isa->value() == "i386") {
-            static map<boost::regex, string> virtual_map = {
-                {boost::regex("VMware"),     string(vm::vmware)},
-                {boost::regex("VirtualBox"), string(vm::virtualbox)},
-                {boost::regex("Parallels"),  string(vm::parallels)},
-                {boost::regex("KVM"),        string(vm::kvm)},
-                {boost::regex("HVM domU"),   string(vm::xen_hardware)},
-                {boost::regex("oVirt Node"), string(vm::ovirt)}
-            };
+        static map<boost::regex, string> virtual_map = {
```

Comment:
What I mean is we could just add an entry for `VMWare` in this list so that it matches both without having to use regexes.  I believe that's the only one that had a legitimate use for using regular expressions in the first place.